operator/v1/types_network.go

```diff
@@ -321,6 +321,11 @@ type OVNKubernetesConfig struct {
 	// not using OVN.
 	// +optional
 	HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+	// enableIPsec enables IPsec encryption on all pod-pod traffic within the cluster.
+	// Default is false.
+	// +optional
+	// +kubebuilder:default:=false
+	EnableIPsec bool `json:"enableIPsec"`
```

Comment:
Is there any chance we're ever going to need any more IPsec configurability? There had been discussion about maybe needing different cryptosystems in some cases?

If we think we might then it would be better to have an `IPsecConfig` struct (like `HybridOverlayConfig` above). If we're pretty sure we won't then the bool is fine (and, worst case, if we unexpectedly added more config later we could add a struct then and deprecate the original bool)

Reply:
:+1

Reply:
This is a good point. The enhancement proposal didn't have this but, since then, I have gotten some feedback of what could potentially get configured. I will update as per your suggestion.

Reply:
@sttts Is 'IPsecConfig' good for this or should I choose something less generic for this type - e.g. OVNKubernetesIPsecConfig?

Reply:
Since this is within the OVNKubernetesConfig, prefixing the struct with that is redundant. `IPsecConfig` is good enough IMO.

Reply:
All these types are defined in the v1 package so just wasnt sure.

Reply:
It's a gray area. But note that what really matters is the *json* not the Go structs. The JSON is the actual API here, and tha'ts namespaced inside the ovnkubernetes block.

Reply:
Good point. We can easily change the Go structs after if there is some kind of namespace collision.